pkg/controller/gkenetworkparamset/gkenetworkparamset_controller.go

```diff
@@ -148,6 +151,22 @@ func (c *Controller) handleErr(err error, key interface{}) {
 	klog.Errorf("Dropping GKENetworkParamSet %q out of the queue: %v", key, err)
 }
 
+// addFinalizerToGKENetworkParamSet add a finalizer to params if it doesnt already exist
+func (c *Controller) addFinalizerToGKENetworkParamSet(ctx context.Context, params *networkv1alpha1.GKENetworkParamSet) error {
+	for _, f := range params.ObjectMeta.Finalizers {
+		if f == GNPFinalizer {
+			return nil
+		}
+	}
+
+	params.ObjectMeta.Finalizers = append(params.ObjectMeta.Finalizers, GNPFinalizer)
+	if err := c.updateGKENetworkParamSet(ctx, params); err != nil {
```

Comment:
lets make sure the update is only done once for whole reconcile function, remove.

Reply:
Done